judge/views/submission.py

```diff
@@ -531,6 +531,21 @@ def get(self, request, *args, **kwargs):
         return super(ForceContestMixin, self).get(request, *args, **kwargs)
 
 
+class UserAllContestSubmissions(ForceContestMixin, AllUserSubmissions):
```

Comment:
Hmm... I think we're missing the actual submission filtering.

Reply:
Which submission filtering are you referring to? `ForceContestMixin` causes `in_contest` to be True, so when `get_queryset()` is called in `AllUserSubmissions`, it will call `SubmissionListBase`, which with `in_contest` as True will filter for only contest submissions, and then `AllUserSubmissions` will filter by the user.

Reply:
Ah, you're correct. Filtering is already done properly in `SubmissionListbase`.